tools/slicec-cs/src/generators/enum_generator.rs

```diff
@@ -233,11 +509,54 @@ fn enum_decoder_extensions(enum_def: &Enum) -> CodeBlock {
         format!(r#"Provides an extension method for decoding a <see cref="{escaped_identifier}" /> using a <see cref="SliceDecoder" />."#),
     ).add_generated_remark("static class", enum_def);
 
-    let underlying_extensions_class = format!(
-        "{}{}Extensions",
-        enum_def.cs_identifier(Case::Pascal),
-        cs_type.to_cs_case(Case::Pascal),
-    );
+    let decode_body = if enum_def.is_mapped_to_cs_enum() {
+        let underlying_extensions_class = format!(
+            "{}{}Extensions",
+            enum_def.cs_identifier(Case::Pascal),
+            cs_type.to_cs_case(Case::Pascal),
+        );
+
+        format!(
+            "{underlying_extensions_class}.As{identifier}({decode_enum});",
+            identifier = enum_def.cs_identifier(Case::Pascal),
+            decode_enum = match &enum_def.underlying {
+                Some(underlying) => format!("decoder.Decode{}()", underlying.definition().type_suffix()),
+                _ => "decoder.DecodeSize()".to_owned(), // Slice1 only
+            },
+        )
+    } else {
+        let mut cases = CodeBlock::default();
+
+        for enumerator in enum_def.enumerators() {
+            let enumerator_class = format!(
+                "{escaped_identifier}.{enumerator_name}",
+                enumerator_name = enumerator.escape_identifier()
```

Comment:
Trailing comma after each parameter in Rust, even the last one.
```suggestion
                enumerator_name = enumerator.escape_identifier(),
```